src/analysis/asymptotics/asymptotics.lean

```diff
@@ -1247,6 +1247,26 @@ end exists_mul_eq
 
 /-! ### Miscellanous lemmas -/
 
+theorem is_O_of_div_is_bounded_under {α : Type*} {l : filter α}
+  {f g : α → 𝕜} (hgf : ∀ᶠ x in l, g x = 0 → f x = 0)
+  (h : is_bounded_under (≤) l (λ x, ∥f x / g x∥)) :
+  is_O f g l :=
```

Comment:
Even without `hgf`, we have `is_O f g l` implies `h` (because of how division by zero is defined in mathlib, the reverse direction doesn't need `hgf` at all), so I added this as another lemma as well as making this an `iff` lemma. 

I'm not sure though if `is_O_of_div_tendsto_nhds` can be made into an `iff` statement, since I think it would require some statement like `tendsto_nhds_of_is_bounded_under` which I don't think would be true in general.